python/tank/bootstrap/manager.py

```diff
@@ -646,16 +649,23 @@ def _cache_apps(self, tk, progress_callback, do_post_install=False):
         descriptors = []
         for env_name in pc.get_environments():
 
-            env_obj = pc.get_environment(env_name)
+            # Select the site or project environment based on the configuration project id.
+            if (not config_project_id and env_name == "site") or (config_project_id and env_name != "site"):
```

Comment:
Unfortunately, the `site` env name is just convention and not hard coded. It's all decided inside the `pick_environment` hook so somone could decide to return `foo` for the site config. the only way to find out is to execute the pick environment hook. Is that possible at this stage you think? Perhaps this method: https://github.com/shotgunsoftware/tk-core/blob/master/python/tank/platform/engine.py#L2789